package/ios/RNSkia-iOS/SkiaMetalSurfaceFactory.h

```diff
@@ -4,18 +4,22 @@
 #pragma clang diagnostic ignored "-Wdocumentation"
 
 #import "include/core/SkCanvas.h"
+#import <CoreMedia/CMSampleBuffer.h>
+#import <CoreVideo/CVMetalTextureCache.h>
 #import <include/gpu/GrDirectContext.h>
+#import <memory>
 
 #pragma clang diagnostic pop
 
 using SkiaMetalContext = struct SkiaMetalContext {
+  id<MTLDevice> device = nullptr;
   id<MTLCommandQueue> commandQueue = nullptr;
   sk_sp<GrDirectContext> skContext = nullptr;
 };
 
 class ThreadContextHolder {
 public:
-  static thread_local SkiaMetalContext ThreadSkiaMetalContext;
+  static const std::unique_ptr<SkiaMetalContext>& getThreadSpecificSkiaContext();
```

Comment:
this doesn't appear to fix the issue for me.
My test scenario is different as I create offscreen surfaces before testing the platform buffers (so I initialize things you might not have on your side).
Also if there is a change to be made here, should be done as a standalone change first?